karrot/activities/receivers.py

```diff
@@ -64,6 +67,29 @@ def remove_activity_participant_from_conversation(sender, instance, **kwargs):
     conversation.leave(user)
 
 
+@receiver(post_save, sender=ActivityParticipant)
+def schedule_activity_reminder(sender, instance, **kwargs):
+    participant = instance
+    if participant.reminder_task_id:
+        return
+
+    activity = participant.activity
+    remind_at = activity.date.start - timedelta(hours=3)
+    task = tasks.activity_reminder.schedule(
+        (participant.id, ),
+        eta=remind_at,
+    )
+    participant.reminder_task_id = task.id
+    participant.save()
```

Comment:
This might be nice to save without triggering the receiver signal again... I accidently got it into a loop once when I changed the task to clear the `reminder_task_id` after it ran, but then it endlessly rescheduled itself... oops! A bit too easy to trigger that...